test/SlowTests/Server/Documents/ETL/Elasticsearch/ElasticsearchEtlTests.cs

```diff
@@ -0,0 +1,822 @@
+                                ElasticIndexes =
+                                {
+                                    new ElasticsearchIndex {IndexName = "Orders", IndexIdProperty = "Id"},
+                                    new ElasticsearchIndex {IndexName = "OrderLines", IndexIdProperty = "OrderId"},
+                                    new ElasticsearchIndex {IndexName = "NotUsedInScript", IndexIdProperty = "OrderId"},
+                                },
+                                Transforms =
+                                {
+                                    new Transformation() {Collections = {"Orders"}, Name = "OrdersAndLines", Script = defaultScript + "output('test output')"}
+                                }
+                            }
+                        }, database, database.ServerStore, context);
+
+                    Assert.Equal(0, result.TransformationErrors.Count);
+
+                    Assert.Equal(2, result.Summary.Count);
+
+                    var orderLines = result.Summary.First(x => x.IndexName == "orderlines");
+
+                    Assert.Equal(1, orderLines.Commands.Length); // delete
+
+                    var orders = result.Summary.First(x => x.IndexName == "orders");
+
+                    Assert.Equal(1, orders.Commands.Length); // delete
+                }
+                
+                using (var session = store.OpenAsyncSession())
+                {
+                    Assert.NotNull(session.Query<Order>("orders/1-A"));
+                }
+            }
+        }
+
+        protected void SetupElasticEtl(DocumentStore store, string script, IEnumerable<string> collections = null, bool applyToAllDocuments = false)
+        {
+            var connectionStringName = $"{store.Database}@{store.Urls.First()} to ELASTIC";
+
+            AddEtl(store,
+                new ElasticsearchEtlConfiguration()
+                {
+                    Name = connectionStringName,
+                    ConnectionStringName = connectionStringName,
+                    ElasticIndexes =
+                    {
+                        new ElasticsearchIndex {IndexName = "Orders", IndexIdProperty = "Id"},
+                        new ElasticsearchIndex {IndexName = "OrderLines", IndexIdProperty = "OrderId"},
+                        new ElasticsearchIndex {IndexName = "Users", IndexIdProperty = "UserId"},
+                    },
+                    Transforms =
+                    {
+                        new Transformation()
+                        {
+                            Name = $"ETL : {connectionStringName}",
+                            Collections = new List<string>(collections),
+                            Script = script,
+                            ApplyToAllDocuments = applyToAllDocuments
+                        }
+                    }
+                },
+                new ElasticsearchConnectionString {Name = connectionStringName, Nodes = new[] {"http://localhost:9200"}});
```

Comment:
how are these tests going to work in our CI ?
don't we need to spin up elastic instance in `localhost:9200` ?
or am I missing something ?

Reply:
We'll take care of setting up our CI accordingly on our own.